cmd/poller/poller.go

```diff
@@ -678,10 +678,15 @@ func (p *Poller) loadCollectorObject(ocs []objectCollector) error {
 					Msg("abort collector")
 				break
 			}
-			logger.Warn().Err(err).
-				Str("collector", oc.class).
-				Str("object", oc.object).
-				Msg("init collector-object")
+			if oc.class == "Zapi" && oc.object == "Status_7mode" {
```

Comment:
This change can ignore the warning for 7mode as well. 
But, It would be less chances for this `file not found` in 7mode.


After deleted the file locally.
Without this change:
```
hardikl@hardikl-mac-2 harvest % ./bin/poller -p cluster-51 --promPort 14010 -c Zapi 
2022-12-09T21:06:45+05:30 INF ./poller.go:202 > Init Poller=cluster-51 configPath=./harvest.yml logLevel=info version="harvest version 22.12.0920-v22.11.1 (commit 1ad9bdf7) (build date 2022-12-09T20:31:40+0530) darwin/amd64\n"
2022-12-09T21:06:45+05:30 INF ./poller.go:239 > started in foreground Poller=cluster-51 pid=14556
2022-12-09T21:06:49+05:30 INF collector/helpers.go:134 > best-fit template Poller=cluster-51 collector=Zapi:Disk path=conf/zapi/7mode/8.6.0/disk.yaml v=8.2.5
2022-12-09T21:06:51+05:30 WRN ./poller.go:684 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=NtpServer
2022-12-09T21:06:52+05:30 INF collector/helpers.go:134 > best-fit template Poller=cluster-51 collector=Zapi:Qtree path=conf/zapi/7mode/8.6.0/qtree.yaml v=8.2.5
2022-12-09T21:06:55+05:30 WRN ./poller.go:684 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=SecurityAccount
2022-12-09T21:06:57+05:30 WRN ./poller.go:684 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=SecurityLogin
2022-12-09T21:06:58+05:30 INF collector/helpers.go:134 > best-fit template Poller=cluster-51 collector=Zapi:Shelf path=conf/zapi/7mode/8.6.0/shelf.yaml v=8.2.5
2022-12-09T21:07:01+05:30 INF collector/helpers.go:134 > best-fit template Poller=cluster-51 collector=Zapi:Status path=conf/zapi/7mode/8.6.0/status.yaml v=8.2.5
2022-12-09T21:07:02+05:30 WRN ./poller.go:684 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=AggregateEfficiency
2022-12-09T21:07:03+05:30 INF collector/helpers.go:134 > best-fit template Poller=cluster-51 collector=Zapi:Node path=conf/zapi/7mode/8.6.0/node.yaml v=8.2.5
2022-12-09T21:07:05+05:30 WRN ./poller.go:684 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=SnapshotPolicy
2022-12-09T21:07:06+05:30 WRN ./poller.go:684 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=Status_7mode
2022-12-09T21:07:08+05:30 WRN ./poller.go:684 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=Support
2022-12-09T21:07:09+05:30 WRN ./poller.go:684 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=SVM
2022-12-09T21:07:10+05:30 INF collector/helpers.go:134 > best-fit template Poller=cluster-51 collector=Zapi:Aggregate path=conf/zapi/7mode/8.6.0/aggr.yaml v=8.2.5
2022-12-09T21:07:12+05:30 WRN ./poller.go:684 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=ClusterPeer
2022-12-09T21:07:13+05:30 INF collector/helpers.go:134 > best-fit template Poller=cluster-51 collector=Zapi:Lun path=conf/zapi/7mode/8.6.0/lun.yaml v=8.2.5
```


With this change:
```
hardikl@hardikl-mac-2 harvest % ./bin/poller -p cluster-51 --promPort 14010 -c Zapi
2022-12-09T21:08:35+05:30 INF ./poller.go:202 > Init Poller=cluster-51 configPath=./harvest.yml logLevel=info version="harvest version 22.12.0921-v22.11.1 (commit 3f2b08ce) (build date 2022-12-09T21:07:42+0530) darwin/amd64\n"
2022-12-09T21:08:35+05:30 INF ./poller.go:239 > started in foreground Poller=cluster-51 pid=14982
2022-12-09T21:08:39+05:30 INF collector/helpers.go:134 > best-fit template Poller=cluster-51 collector=Zapi:Aggregate path=conf/zapi/7mode/8.6.0/aggr.yaml v=8.2.5
2022-12-09T21:08:41+05:30 WRN ./poller.go:688 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=NtpServer
2022-12-09T21:08:42+05:30 WRN ./poller.go:688 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=SecurityAuditDestination
2022-12-09T21:08:44+05:30 WRN ./poller.go:688 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=SecurityCert
2022-12-09T21:08:46+05:30 WRN ./poller.go:688 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=AggregateEfficiency
2022-12-09T21:08:48+05:30 WRN ./poller.go:688 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=SecurityAccount
2022-12-09T21:08:49+05:30 INF collector/helpers.go:134 > best-fit template Poller=cluster-51 collector=Zapi:Subsystem path=conf/zapi/7mode/8.6.0/subsystem.yaml v=8.2.5
2022-12-09T21:08:50+05:30 INF collector/helpers.go:134 > best-fit template Poller=cluster-51 collector=Zapi:Volume path=conf/zapi/7mode/8.6.0/volume.yaml v=8.2.5
2022-12-09T21:08:52+05:30 INF collector/helpers.go:134 > best-fit template Poller=cluster-51 collector=Zapi:Disk path=conf/zapi/7mode/8.6.0/disk.yaml v=8.2.5
2022-12-09T21:08:53+05:30 INF collector/helpers.go:134 > best-fit template Poller=cluster-51 collector=Zapi:Node path=conf/zapi/7mode/8.6.0/node.yaml v=8.2.5
2022-12-09T21:08:54+05:30 INF collector/helpers.go:134 > best-fit template Poller=cluster-51 collector=Zapi:Qtree path=conf/zapi/7mode/8.6.0/qtree.yaml v=8.2.5
2022-12-09T21:08:57+05:30 INF collector/helpers.go:134 > best-fit template Poller=cluster-51 collector=Zapi:Shelf path=conf/zapi/7mode/8.6.0/shelf.yaml v=8.2.5
2022-12-09T21:09:00+05:30 WRN ./poller.go:688 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=SnapshotPolicy
2022-12-09T21:09:01+05:30 INF collector/helpers.go:134 > best-fit template Poller=cluster-51 collector=Zapi:Status path=conf/zapi/7mode/8.6.0/status.yaml v=8.2.5
2022-12-09T21:09:02+05:30 WRN ./poller.go:688 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=ClusterPeer
2022-12-09T21:09:04+05:30 INF collector/helpers.go:134 > best-fit template Poller=cluster-51 collector=Zapi:Lun path=conf/zapi/7mode/8.6.0/lun.yaml v=8.2.5
2022-12-09T21:09:05+05:30 WRN ./poller.go:688 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=Security
2022-12-09T21:09:06+05:30 WRN ./poller.go:688 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=SecurityLogin
2022-12-09T21:09:08+05:30 WRN ./poller.go:688 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=Sensor
2022-12-09T21:09:09+05:30 WRN ./poller.go:688 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=SnapMirror
2022-12-09T21:09:10+05:30 WRN ./poller.go:688 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=Support
2022-12-09T21:09:12+05:30 WRN ./poller.go:688 > init collector-object error="unable to import template=[] no best-fit template found" Poller=cluster-51 collector=Zapi object=SVM
2022-12-09T21:09:12+05:30 INF ./poller.go:371 > Autosupport scheduled. Poller=cluster-51 asupSchedule=24h
2022-12-09T21:09:12+05:30 INF ./poller.go:380 > poller start-up complete Poller=cluster-51
2022-12-09T21:09:12+05:30 INF prometheus/httpd.go:29 > http server listen Poller=cluster-51 addr= exporter=prometheus2 port=14010
2022-12-09T21:09:12+05:30 INF ./poller.go:528 > updated status, up collectors: 9 (of 9), up exporters: 1 (of 1) Poller=cluster-51
```

Reply:
Yes, I looked at making the change lower in the stack but wasn't able to find a place to suppress without requiring more significant changes. And I agree, I don't love that this commit hides a valid log msg for 7mode. I'll look again. Other ideas rejected:
- the place the poller handles the error, it does not know if it's talking to cdot or 7-mode and it's not really the pollers responsibility anyway. The poller is responsible for creating the new collector and excepts a collector or error.
- a blank template for cdot, won't work either since we want to log that as a warning too 
- detecting the problem when the template is read in helpers.go where we know if we're talking to a cdot or 7mode cluster, won't work because at best, we can return a nil node which will also be logged as a warning

Instead of a workaround we could be explicit by extending the template
- add a `skip` key/val to the template, but meh
- other template changes meh

I'm going to try changing helper.go to return an error indicating version mismatch and handle that in poller around the same place as the current commit's check.

With 167ec9bac21ac3d7d76801b9dbbc6830645f0d54  if I delete `status_7mode.yaml` and run a 7mode poller it does what we want, prints a warning while suppressing it for cdot.

```
2022-12-09T14:17:05-05:00 WRN poller/poller.go:687 > init collector-object error="unable to import template=[] no best-fit template found" Poller=v-72 collector=Zapi object=Status_7mode
```